scene/main/viewport.cpp

```diff
@@ -672,17 +672,20 @@ void Viewport::_notification(int p_what) {
 		} break;
 	}
 }
-
+constexpr char const *const LEGACY_INPUT_SETTING = "physics/legacy_picking_input_processing";
 void Viewport::_process_picking() {
 	if (!is_inside_tree()) {
 		return;
 	}
 	if (!physics_object_picking) {
 		return;
 	}
-	if (Object::cast_to<Window>(this) && Input::get_singleton()->get_mouse_mode() == Input::MOUSE_MODE_CAPTURED) {
+	auto const ps = ProjectSettings::get_singleton();
```

Comment:
```suggestion
	ProjectSettings *ps = ProjectSettings::get_singleton();
```
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)